src/daos_vol_file.c

```diff
@@ -2990,8 +2986,9 @@ H5_daos_file_delete_status_bcast_comp_cb(tse_task_t *task, void H5VL_DAOS_UNUSED
     /* Handle errors in bcast task.  Only record error in udata->req_status if
      * it does not already contain an error (it could contain an error if
      * another task this task is not dependent on also failed). */
-    if(task->dt_result < -H5_DAOS_PRE_ERROR
-            && udata->req->status >= -H5_DAOS_SHORT_CIRCUIT) {
+    if((task->dt_result < -H5_DAOS_PRE_ERROR
+            && udata->req->status >= -H5_DAOS_SHORT_CIRCUIT) ||
+            ( *((herr_t *)udata->buffer) < 0)) {
```

Comment:
My preference would be for this check to be in a different block, this is sort of a one-off change to boilerplate code, and a more specific error message for the case where the lead rank failed would be helpful anyways

Reply:
Also I think "task->dt_result" would have the wrong value in that case

Reply:
Sshould be fixed now